lib/src/executor_impl.dart

```diff
@@ -68,7 +68,10 @@ class _Executor implements Executor {
     _running.remove(item);
     _trigger();
     item.done.complete();
-    return item.result.future;
+    return await item.result.future
+        // Nullable R is used to allow using catchError with null output, so
+        // we must convert R? into R for the caller
+        .then((v) => v ?? v as R);
```

Comment:
This feels like something that shouldn't really happen, or rather: something that we shouldn't write. I'd think `(v) => v as R` would be sufficient, but then it would be a no-op, wouldn't it?

Reply:
result.future is `R?` so removing the `then` would not compile. I've updated the code with simply `v as R`